File: backend/app/schemas/logs.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MailFetchLogPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    trace_id: str
    user_id: int | None
    api_key_id: int | None
    mail_account_id: int | None
    email: str
    mailbox: str
    operation: str
    source_protocol: str | None
    status: str
    error_code: str | None
    error_message: str | None
    duration_ms: int | None
    created_at: datetime
    updated_at: datetime


class AuditLogPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    actor_user_id: int | None
    action: str
    target_type: str
    target_id: str
    metadata_json: dict | None
    ip_address: str | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/api/logs.ts
import { http } from './http'

export interface MailFetchLog {
  id: number
  trace_id: string
  user_id: number | null
  api_key_id: number | null
  mail_account_id: number | null
  email: string
  mailbox: string
  operation: string
  source_protocol: string | null
  status: string
  error_code: string | null
  error_message: string | null
  duration_ms: number | null
  created_at: string
  updated_at: string
}

export interface MailFetchLogFilters {
  email?: string
  status?: string
}

export function fetchMailFetchLogs(filters: MailFetchLogFilters = {}) {
  const params = new URLSearchParams()
  if (filters.email) params.set('email', filters.email)
  if (filters.status) params.set('status', filters.status)
  const qs = params.toString()
  return http.get<MailFetchLog[]>(`/api/logs/mail-fetch${qs ? `?${qs}` : ''}`)
}
